src/cheri_insts.sail

```diff
@@ -229,6 +231,79 @@ function clause execute (CGetOffset(rd, cs1)) = {
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the **upper half** of capability
+ * register *cs1*.
+ *
+ * - The bits returned here are of the **in-register** form of the capability,
+ *   which differs from the in-memory form by the XOR of [null_cap_bits].  As
+ *   there are set bits in the upper half of this value, [CGetHigh] will
+ *   differ from storing *cs1* to memory and reading back the high word.
+ */
+function clause execute (CGetHigh(rd, cs1)) = {
+  let capVal : Capability = C(cs1);
+  X(rd) = truncate(capToBits(capVal) >> sizeof(xlen), sizeof(xlen));
+  RETIRE_SUCCESS
+}
+
+union clause ast = CSetHigh : (regidx, regidx, regidx)
+/*!
+ * Capability register *cd* comes to hold the capability from *cs1* with its
+ * upper bits replaced with the value in the integer register *rs1*.  The tag
+ * of *cd* is cleared.
+ *
+ * - *rs* holds the in-register form of capability bits.  These will be xored
```

Comment:
Previous comment mostly applies to the writing side, but I think we should use the in-memory representation for both.